In [1]:
import seaborn as sns
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
from google.colab import drive

drive.mount("/content/drive")

Mounted at /content/drive


## Подгрузим данные

In [3]:
medicine_data = pd.read_csv(
    filepath_or_buffer="/content/drive/MyDrive/Colab Notebooks/Hackathon/mock-med-data/medicine_data.csv"
)
medicine_data.head(4)

,код_препарата,дозировка,Торговое название,стоимость,Полное_название
0,200052200,1.5,Галоперидол,12.98,"Галоперидол, 1,5 мг № 50"
1,200052400,5,Галоперидол,15.51,"Галоперидол, 5 мг № 50, таблетки"
2,200059000,3.5,Манинил,93.42,"Манинил, 3,5 мг № 120, таблетки"
3,200165100,50,Пури-Нетол,525.48,"Пури-Нетол, 50 мг № 25, таблетки"


## Наличие Nan

In [36]:
print(medicine_data.isna().sum())
print("\n")

print(f"{(medicine_data.isna().sum() / len(medicine_data)) * 100} %(percentage of nan's)", "\n")

код_препарата           0
дозировка             116
Торговое название       0
стоимость            2201
Полное_название         0
dtype: int64


код_препарата         0.000000
дозировка             3.418803
Торговое название     0.000000
стоимость            64.868848
Полное_название       0.000000
dtype: float64 %(percentage of nan's) 



### Информация по типам

In [5]:
medicine_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3393 entries, 0 to 3392
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   код_препарата      3393 non-null   int64  
 1   дозировка          3277 non-null   object 
 2   Торговое название  3393 non-null   object 
 3   стоимость          1192 non-null   float64
 4   Полное_название    3393 non-null   object 
dtypes: float64(1), int64(1), object(3)
memory usage: 132.7+ KB


## Форма датасета

In [6]:
medicine_data.shape

(3393, 5)

## Количество уникальных лэйблов

In [8]:
for col in medicine_data.columns:
    print(f"{col} : ", {len(pd.unique(medicine_data[col])): pd.unique(medicine_data[col])[:10]}, "\n")

код_препарата :  {3393: array([200052200, 200052400, 200059000, 200165100, 200194100, 200214800,
       200247900, 200331600, 200411000, 200531000])} 

дозировка :  {325: array(['1.5', '5', '3.5', '50', '0.01', '400', '200', '0.5', '0.25',
       '100'], dtype=object)} 

Торговое название :  {2028: array(['Галоперидол', 'Манинил', 'Пури-Нетол', 'Нифедипин', 'Пирацетам',
       'Теотард', 'НовоНорм', 'Тимолол-АКОС', 'Мексикор', 'Дюфастон'],
      dtype=object)} 

стоимость :  {1051: array([ 12.98,  15.51,  93.42, 525.48,  12.6 ,  31.02,  60.09, 198.32,
       106.7 , 286.09])} 

Полное_название :  {3382: array(['Галоперидол, 1,5 мг № 50', 'Галоперидол, 5 мг № 50, таблетки',
       'Манинил, 3,5 мг № 120, таблетки',
       'Пури-Нетол, 50 мг № 25, таблетки',
       'Нифедипин, 10 мг № 50, таблетки',
       'Пирацетам, 400 мг № 60, таблетки',
       'Теотард, 200 мг № 40, капсулы', 'НовоНорм, 0,5 мг № 30, таблетки',
       'Тимолол-АКОС, 0,25% № 1, капли глазные',
       'Мексикор, 100 мг

## Самые часто встречающиеся значения

In [40]:
for idx, col in enumerate(medicine_data.columns):
    if idx == 0 or idx == 3:
        continue
    print(medicine_data.iloc[:, idx].value_counts(ascending=False)[0:10], "\n")

дозировка
100    294
10     272
50     212
5      162
20     133
25     129
250    124
200    117
500    116
150     92
Name: count, dtype: int64 

Торговое название
Гемофил М                                        12
Метотрексат                                      12
Салофальк                                        10
Эральфон                                         10
Ксарелто                                         10
Методжект                                        10
Иглы (пр-ва SFM "СФМ Госпитал Продактс ГмбХ")     9
Иглы KD-PENOFINE                                  9
Эниксум                                           8
Метортрит                                         8
Name: count, dtype: int64 

Полное_название
Кортексин, 5 мг № 10, лиофилизат для приготовления раствора для внутримышечного введения          2
Пульмикорт Турбухалер, 200 мкг/доза 100 доз № 1, порошок для ингаляций дозированный               2
Метотрексат, 2,5 мг № 50, таблетки покрытые оболочкой                 

In [41]:
medicine_data.loc[medicine_data["стоимость"].notna(), "стоимость"].value_counts()

,count
стоимость,
277.50,5
84457.20,5
270.00,5
81.00,4
11431.20,4
...,...
374.64,1
749.28,1
390456.00,1


## Посмотрим на строки с часто встерчаемые названиями

In [42]:
medicine_data[medicine_data["Торговое название"] == "Гемофил М"]

,код_препарата,дозировка,Торговое название,стоимость,Полное_название
1348,201800161,(850-1240 МЕ) 970,Гемофил М,NaN,"Гемофил М, 850-1240 МЕ (970 МЕ) № 1, лиофилиза..."
2483,201700081,(850-1240 МЕ) 1010,Гемофил М,NaN,"Гемофил М, 850-1240 МЕ (1010 МЕ) № 1, лиофилиз..."
2484,201700082,(850-1240 МЕ) 966,Гемофил М,NaN,"Гемофил М, 850-1240 МЕ (966 МЕ) № 1, лиофилиза..."
2485,201700083,(850-1240 МЕ) 939,Гемофил М,NaN,"Гемофил М, 850-1240 МЕ (939 МЕ) № 1, лиофилиза..."
2486,201700084,(850-1240 МЕ) 867,Гемофил М,NaN,"Гемофил М, 850-1240 МЕ (867 МЕ) № 1, лиофилиза..."
2487,201700085,(850-1240 МЕ) 1040,Гемофил М,NaN,"Гемофил М, 850-1240 МЕ (1040 МЕ) № 1, лиофилиз..."
2752,201800096,(850-1240 МЕ) 1020,Гемофил М,NaN,"Гемофил М, 850-1240 МЕ (1020 МЕ) № 1, лиофилиз..."
2753,201800097,(850-1240 МЕ) 964,Гемофил М,NaN,"Гемофил М, 850-1240 МЕ (964 МЕ) № 1, лиофилиза..."
2832,201400166,800-1400,Гемофил М,NaN,"Гемофил М, 800-1400 МЕ (1090 МЕ) № 1, лиофилиз..."
3014,201100057,850-1240,Гемофил М,NaN,"Гемофил М, 850-1240 МЕ (1060 МЕ) № 1, лиофилиз..."


## Выводы:
  - 64 % пропусков в столбце `стоимость` - нужно настроить парсинг с сайтов цен на эти препараты
  - 3 столбца полностью известны (`код препарата`, `Торговое название`, `Полное название`)  
  - 3393 строки, 5 столбцов
  - `дозировка` имеет 3 % процента пропущенных значений поэтому имеет смысл также допарсить  или импутацией воспользоваться(лучше не стоит).
  